### ISSUE TITLE
Image relaods as user steps through images

### ISSUE THREAD

**Issue by nerdykit** [netPhotoGraphics/netPhotoGraphics#1165]
As a user steps through a group of images by pressing the <next> link at the top of the image page, each new image page is presented with a brief flash. It's almost as if the image appears, disappears and reappears in about a half-second or less (might be longer for viewing from remote computers...I'm on the same subnet as my server so speeds are very fast).  This behavior has something to do with panorama plugin because it is not there if the plugin is disabled and it was not there before the panorama plugin was being used.  It's almost as if the image is being loaded twice, once by the page before it encounters the panorama plugin and then the image is loaded again by panorama.

If I change from my slightly customized version of theme basic to the distributed version of basic, I can see that each image loads at one ratio (about 2 by 3 ratio) then reloads in a different ratio that is closer to square.  I think that what I'm seeing using my own version of basic theme is that the two different loads are the same size so it looks like a "flash" of the image but if I use the distributed version of the basic theme, the size change as each new image page is presented is quite easily noticed.

Send me a message if you would like a link to the site to view this for yourself.  You will also be able to tell me if what is a brief "flash" of the image for me, is somewhat longer for a remote user.  I can't leave the site set to distributed version of basic theme so that you can see the unwanted effect more clearly because that messes up all of the other albums since their images are all cached at incorrect sizes for the use of the distributed version of basic theme.

**Comment by sbillard** [comment]
Here is a link to my test site panoramas (actually the first image.) 
What I see is that the image is loaded and shows full size then is 
resized into the panorama. Is that what you are seeing? On my local test 
site there is an almost unnoticeable flash. I am not sure why there is a 
difference since all the rendering is being done on my client PC. But 
maybe the load time makes a difference.

There might be a way to keep the image hidden until paver has dealt with 
it. Have to look at its options.


-Stephen

**Comment by sbillard** [comment]
Sorry, forgot to include the link: 
https://testalbum-2.sbillard.org/test_two/panoramas/hurtigruten-south-pano3-KD1_4672-KD1_4680-wb.jpg.htm


-Stephen

**Comment by nerdykit** [comment]
OK, now I'll take a look.  I was just typing a note about the missing link.

Actually the reload is noticed on both panoramic images and on standard images.  My albums are a mixture of almost all regular images with a few panoramas in there.

**Comment by nerdykit** [comment]
Yes, it's definitely the panorama reload that we're seeing.  For conventional images (you could put a couple into your test site just for the comparison) in my case both original image and the panorama reload are the same size so it's only a sort of image flicker.  But where the images are different sizes the shift is very noticeable.  Keeping the original image hidden until panorama has taken over (if enabled) would probably deal with this.

**Comment by sbillard** [comment]
Give this a try. I can hide the image until paver is ready. There is a 
bit of a "flash" since the place where the image is will be blank until 
paver is ready.


-Stephen

**Comment by nerdykit** [comment]
Did you mean to check out your test site?  It looks about the same but maybe I should try editing my site so that I can see the change in an environment I'm more familiar with.

**Comment by sbillard** [comment]
Sorry, I replied by email and attached a test version of the plugin. I'll stick it here since it apparently did not come through.
[panorama.zip](https://github.com/netPhotoGraphics/netPhotoGraphics/files/4625257/panorama.zip)

**Comment by nerdykit** [comment]
Got it via github.  Gmail doesn't like zip files containing javascript.  Will install and report back.

**Comment by nerdykit** [comment]
After testing.  Looks really good.  I've no idea if it looks very different if the viewer isn't sitting right beside the web server but we'll probably have to wait until the distancing ends until I can get over to a friend's place to take a look at how the site looks for remote users.  But for now it works just fine.  I think that should be your production version of panorama.

**Comment by nerdykit** [comment]
Leaving my desk for a while.  In Canada everything stops for tea!

**Comment by sbillard** [comment]
I have put it up on my test site, so you can see how it looks from afar.

> Leaving my desk for a while. In Canada everything stops for tea!

I understand. I lived in England for 3 years. It was either tea or the Pub....

**Comment by nerdykit** [comment]
Looks fine.  Before you put this version in place, you got a brief look at the full image at full width and then saw it in "presentation" mode when it reloaded.  Now you just see the finished presentation and it's much better.  The delay can't be more than half a second so it isn't really an issue.  If other users don't like it then they can opt to not enable panorama.  I've no idea how many people are actually wanting to display this sort of image.  I just started taking them when I found that I could take a hand-held series of overlapping shots and then have them stitched seamlessly by the incredible open source Hugin app.
